### PR TITLE
Money.from_cents works with amounts larger than 2**53.

### PR DESCRIPTION
**Pull request by sgnr**
I guess this is kind of a corner case, but why take the risk...

r: @gmalette @davidcornu @boourns

### CODE REVIEW

File: spec/money_spec.rb [hunk 1]
```diff
@@ -175,6 +175,11 @@
     Money.new(16.51).round(1).should == Money.new(16.5)
   end
 
+  it "can be converted form an amount of cents that cannot be represented as an IEEE 754 double precision float" do
```
Comment (davidcornu) [inline spec/money_spec.rb]: form &rarr; from

Comment (sgnr) [inline spec/money_spec.rb]: fixed

**Comment by davidcornu** [comment]
:+1:

File: spec/money_spec.rb [hunk 2]
```diff
@@ -175,6 +175,11 @@
...
+    amount = 2**53 + 1
+    expect(Money.from_cents(amount).cents).to eq(amount)
```
Comment (gmalette) [inline spec/money_spec.rb]: Because many methods use `@value`, the cents are good, but the `.to_s`, `to_json`, `+`, etc, will produce bad results

Comment (gmalette) [inline spec/money_spec.rb]:
```
> m.cents
=> 9007199254740993
irb(main):037:0> m.to_s
=> "90071992547409.94"
```

Comment (sgnr) [inline spec/money_spec.rb]: This PR is about `Money.from_cents`. I'll open another one for `to_s`.

Comment (gmalette) [inline spec/money_spec.rb]: sure

Comment (gmalette) [inline spec/money_spec.rb]: but then what's the point if this PR doesn't have correct values either?

Comment (sgnr) [inline spec/money_spec.rb]: `Money#value` and `Money#cents` is fixed.

Comment (sgnr) [inline spec/money_spec.rb]: It's just `to_s` that still has to be fixed.

**Comment by sgnr** [comment]
`to_s` is pretty tricky to get right and I don't want to risk a bug in Shopify.

**Pull request closed without merging**